src/launch/launcher.rs

```diff
@@ -93,6 +134,31 @@ impl<'a, U: AsRawFd, V: AsRawFd> Launcher<'a, Started, U, V> {
     }
 }
 
+impl<'a, U: AsRawFd, V: AsRawFd> Launcher<'a, SnpStarted, U, V> {
+    /// Encrypt guest SNP data.
+    pub fn snp_update_data(&mut self, data: &[u8], update: SnpUpdate) -> Result<()> {
+        let launch_update_data = SnpLaunchUpdate::new(data, update);
+        let mut cmd = Command::from(self.sev, &launch_update_data);
+        SNP_LAUNCH_UPDATE
+            .ioctl(self.kvm, &mut cmd)
+            .map_err(|e| cmd.encapsulate(e))?;
+
+        Ok(())
```

Comment:
I think this should automatically pin these pages (see #65 for more info). I opened #68 to backport @haraldh's work from sev-iocuddle along with an example of what I'm talking about to implicitly pin the pages.

Feel free to cherry-pick those commits from my branch if you want to make immediate use of them and get them merged as part of this PR. If you end up doing that, be sure to comment on the other PR so we know we can close it.